build-android/app/src/main/java/com/google/firebase/codelab/friendlychat/MainActivity.kt

```diff
@@ -0,0 +1,258 @@
+        // Check if user is signed in.
+        if (auth.currentUser == null) {
+            // Not signed in, launch the Sign In activity
+            startActivity(Intent(this, SignInActivity::class.java))
+            finish()
+            return
+        }
+    }
+
+    public override fun onPause() {
+        adapter.stopListening()
+        super.onPause()
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        adapter.startListening()
+    }
+
+    public override fun onDestroy() {
+        super.onDestroy()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        val inflater = menuInflater
+        inflater.inflate(R.menu.main_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.sign_out_menu -> {
+                signOut()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        Log.d(TAG, "onActivityResult: requestCode=$requestCode, resultCode=$resultCode")
+        if (requestCode == REQUEST_IMAGE) {
+            if (resultCode == RESULT_OK && data != null) {
+                val uri = data.data
+                Log.d(TAG, "Uri: " + uri.toString())
+                val user = auth.currentUser
+                val tempMessage = FriendlyMessage()
+                tempMessage.name = userName
+                tempMessage.photoUrl = userPhotoUrl
+                tempMessage.imageUrl = LOADING_IMAGE_URL
+                db.reference.child(MESSAGES_CHILD).push()
+                    .setValue(tempMessage,
+                        DatabaseReference.CompletionListener { databaseError, databaseReference ->
+                            if (databaseError != null) {
+                                Log.w(
+                                    TAG, "Unable to write message to database.",
+                                    databaseError.toException()
+                                )
+                                return@CompletionListener
+                            }
+
+                            // Build a StorageReference and then upload the file
+                            val key = databaseReference.key
+                            val storageReference = Firebase.storage
+                                .getReference(user!!.uid)
+                                .child(key!!)
+                                .child(uri!!.lastPathSegment!!)
+                            putImageInStorage(storageReference, uri, key)
+                        })
+            }
+        }
+    }
+
+    private fun putImageInStorage(storageReference: StorageReference, uri: Uri?, key: String?) {
+        // First upload the image to Cloud Storage
+        storageReference.putFile(uri!!)
```

Comment:
If we're going to `uri!!` in the first line then we should just make the parameter type non-optional `Uri`